src/Marten/Linq/SqlGeneration/DocumentStatement.cs

```diff
@@ -33,22 +34,35 @@ protected override ISqlFragment buildWhereFragment(IMartenSession session)
             switch (WhereClauses.Count)
             {
                 case 0:
-                    where = Storage.DefaultWhereFragment();
```

Comment:
Suggestion: you can refactor this by first doing `var wheres = WhereClauses.Select(x => parser.Build(x)).ToArray();` and then do a switch case on `wheres`. It will reduce a bit of clutter. Thoughts?

Reply:
Good point. I'll push an update.